Mathlib/RingTheory/Localization/Pi.lean

```diff
@@ -76,4 +78,31 @@ theorem bijective_lift_piRingHom_algebraMap_comp_piEvalRingHom [IsLocalization M
   (ringEquivOfRingEquiv (M := M) (T := M) _ _ (.refl _) <|
     Submonoid.map_equiv_eq_comap_symm _ _).bijective
 
+open Function in
+lemma surjective_piRingHom_algebraMap_comp_piEvalRingHom_of_nilradical_isMaximal
+(M : Submonoid (Π i, R i)) (h : ∀ i, (nilradical (R i)).IsMaximal)
+  [∀ i, IsLocalization (M.map (Pi.evalRingHom R i)) (S i)] :
+    Surjective (Pi.ringHom (fun i ↦ RingHom.comp (algebraMap (R i) (S i)) (Pi.evalRingHom R i)))
+    := by
```

Comment:
```suggestion
include M in
lemma surjective_piRingHom_algebraMap_comp_piEvalRingHom_of_nilradical_isMaximal
    (h : ∀ i, (nilradical (R i)).IsMaximal) :
    Surjective (Pi.ringHom (fun i ↦ (algebraMap (R i) (S i)).comp (Pi.evalRingHom R i))) := by
```

Reply:
It might also be worth putting one single `open Function` above `surjective_piRingHom_algebraMap_comp_piEvalRingHom_of_nilradical_isMaximal` to avoid having to write `open Function in` twice (though that was my fault:))